soc_026_rw0_global_gender_gap/soc_026_rw0_global_gender_gap.py

```diff
@@ -0,0 +1,191 @@
+    econ_c['year'] = years_list['year'][i] #create a column for year
+    econ_c = econ_c.dropna(subset = ['country']) #drop NA
+    data_clean_subindex.append(econ_c) 
+    
+    edu_a = years_list['dataframes'][i][1].iloc[:, 6:9] #first half
+    edu_a.columns = ['educational_attainment_subindex_rank', 'country', 'educational_attainment_subindex_score']
+    edu_b = years_list['dataframes'][i][1].iloc[:,9:12] #second half
+    edu_b.columns = ['educational_attainment_subindex_rank', 'country', 'educational_attainment_subindex_score']
+    edu_c = pd.concat([edu_a, edu_b]).reset_index(drop=True) #concat
+    edu_c['year'] = years_list['year'][i] #create a column for year
+    edu_c = edu_c.dropna(subset = ['country']) #drop NA
+    data_clean_subindex.append(edu_c)
+    
+    health_a = years_list['dataframes'][i][1].iloc[:, 0:3] #first half
+    health_a.columns = ['health_and_survival_subindex_rank', 'country', 'health_and_survival_subindex_score']
+    health_b = years_list['dataframes'][i][1].iloc[:,3:6] #second half
+    health_b.columns = ['health_and_survival_subindex_rank', 'country', 'health_and_survival_subindex_score']
+    health_c = pd.concat([health_a, health_b]).reset_index(drop=True) #concat
+    health_c['year'] = years_list['year'][i] #create a column for year
+    health_c = health_c.dropna(subset = ['country']) #drop NA
+    data_clean_subindex.append(health_c) 
+    
+    political_a = years_list['dataframes'][i][1].iloc[:, 6:9] #first half
+    political_a.columns = ['political_empowerment_subindex_rank', 'country', 'political_empowerment_subindex_score']
+    political_b = years_list['dataframes'][i][1].iloc[:,9:12] #second half
+    political_b.columns = ['political_empowerment_subindex_rank', 'country', 'political_empowerment_subindex_score']
+    political_c = pd.concat([political_a, political_b]).reset_index(drop=True) #concat
+    political_c['year'] = years_list['year'][i] #create a column for year
+    political_c = political_c.dropna(subset = ['country']) #drop NA
+    data_clean_subindex.append(political_c)
+    
+#    for item in range(len(data_clean_subindex)):
+#        print(item)
+#        while len(data_clean_subindex) > 4:
+#            df_subindex_merged.append(data_clean_subindex.pop(:4))
+
+#Merging all the subindicies for each year. Need to come up with a better way to do this... Tried in the comments above    
+df_subindex_2020 = reduce(lambda left,right: pd.merge(left, right, on=['year', 'country'], how = 'outer'), data_clean_subindex[0:4])
+df_subindex_2021 = reduce(lambda left,right: pd.merge(left, right, on=['year', 'country'], how = 'outer'), data_clean_subindex[4:])
+df_subindex_final = pd.concat([df_subindex_2020, df_subindex_2021])
+
+# Merge main index and subindex
+df_processed = pd.merge(data_main_index, df_subindex_final, on = ['year', 'country'], how = 'outer')
+df_processed = df_processed.replace(',','.', regex=True)
```

Comment:
Pretty much the same changes as above, but I wrote the processing into a function that could be used for each subindex
```suggestion
def process_subindex(subindex, l_r, page):
    if l_r=='left':
        start_i = 0
    else:
        start_i = 6
    cols_sub = [f'{subindex}_subindex_rank', 'country', f'{subindex}_subindex_score']
    df_subindex = pd.DataFrame(columns=cols_sub)
    for year, df_list in df_subindex_dict.items():
        df_a = df_list[page].iloc[:, start_i:start_i+3] #first half
        df_a.columns = cols_sub
        df_b = df_list[page].iloc[:, start_i+3:start_i+6] #second half
        df_b.columns = cols_sub
        df_concat = pd.concat([df_a, df_b]).reset_index(drop=True) #concat
        df_concat['year'] = year #create a column for year
        # replace , with . to properly format decimals in Score column
        df_concat[[cols_sub[2]]] = df_concat[[cols_sub[2]]].replace(',','.', regex=True)
        df_concat = df_concat.dropna(subset = ['country']) #drop NA
        # the source data has mistakenly replaced , with . in some country names
        # if there is a . after the first word of a country name, replace it with a comma
        df_concat['country'] = df_concat['country'].apply(lambda x: x.replace(str(x).split(' ')[0], str(x).split(' ')[0].replace('.', ',')))
        # replace inconsistent country names
        for country in name_mapping.keys():
            df_concat['country'] = np.where(df_concat['country']==country, name_mapping[country], df_concat['country'])
        # drop asterisks from new countries in country column
        df_concat[['country']] = df_concat[['country']].replace('\*','', regex=True)
        df_subindex = pd.concat([df_subindex, df_concat]).reset_index(drop=True)
    return df_subindex


econ_df = process_subindex('economic_participation_and_opportunity', l_r='left', page=0)
edu_df = process_subindex('educational_attainment', l_r='right', page=0)
health_df = process_subindex('health_and_survival', l_r='left', page=1)
political_df = process_subindex('political_empowerment', l_r='right', page=1)

# define the list of dataframes to merge
merge_list = [econ_df, edu_df, health_df, political_df]
df_processed = data_main_index.copy()
for df in merge_list:
    df_processed = pd.merge(df_processed, df, on = ['year', 'country'], how = 'outer')
```